### PR TITLE
Issue #226: Don't strip units off dimensions of non-length

### PR DESCRIPTION
**Pull request by Wizza**
Fix for #226 and #229.

This fix will stop dimensions of non-length from losing their unit type. If these units are stripped off, this can cause some browsers to show the css property as invalid.

E.g
```
transition-duration: 0s;
```
Becomes
```
transition-duration: 0; // In Chrome this will be shown as an invalid property
```

Also removes tests for #82 and #100.

### CODE REVIEW

**Comment by nikrolls** [comment]
:+1: Sorely needed, as the previous patches around this behaviour were very short-sighted and only fixed the symptom rather than the true problem.

**Comment by nikrolls** [comment]
Any update on getting this merged?

File: src/compressor.shared.js [hunk 1]
```diff
@@ -686,14 +686,12 @@ CSSOCompressor.prototype.compressFunctionColor = function(token) {
...
 CSSOCompressor.prototype.compressDimension = function(token) {
-    var declaration;
     if (token[2][2] === '0') {
-        if (token[3][2] === 's' && (declaration = this.findDeclaration(token))) {
-            var declName = declaration[2][2][2];
-            if  (declName === '-moz-transition') return; // https://github.com/css/csso/issues/82
-            if  (declName === '-moz-animation' || declName === 'animation') return; // https://github.com/css/csso/issues/100
-        }
-        return token[2];
+      if (['deg','grad','rad','turn','s','ms','Hz','kHz','dpi','dpcm','dppx'].indexOf(token[3][2]) >= 0) {
```
Comment (snater) [inline src/compressor.shared.js]: It seems that this array should be in constant

**Comment by snater** [comment]
+1 for merging, very annoying bug

**Comment by stenmuchow** [comment]
+1 merge!!!

**Comment by lahmatiy** [comment]
Please, move units array outside functions (as constant).

**Pull request merged**